src/GraphQl/Type/TypeBuilder.php

```diff
@@ -84,80 +85,84 @@ public function getResourceObjectType(?string $resourceClass, ResourceMetadataCo
             $shortName .= 'Data';
         }
 
-        if ($this->typesContainer->has($shortName)) {
-            $resourceObjectType = $this->typesContainer->get($shortName);
-            if (!($resourceObjectType instanceof ObjectType || $resourceObjectType instanceof NonNull)) {
-                throw new \LogicException(sprintf('Expected GraphQL type "%s" to be %s.', $shortName, implode('|', [ObjectType::class, NonNull::class])));
+        $resourceObjectType = null;
+        if (!$this->typesContainer->has($shortName)) {
+            $ioMetadata = $input ? $operation->getInput() : $operation->getOutput();
```

Comment:
the content of this `if` is quite long, what about moving it to a private function?

Reply:
Not sure which `if`-statement you're exactly referring to here?

Reply:
94-162 right now?

Reply:
I see, i've updated the code by extracting that part into a private function.